### ISSUE TITLE
Is there a version for the export of PSK and PSA files?

### ISSUE THREAD

**Issue by GitHubCrazy1** [Befzz/blender3d_import_psk_psa#45]
Just wondering, is there a version with the ability to export PSK and PSA files? I think the older version only works on 2.6 or earlier versions of Blender. Thanks,

**Comment by mirh** [comment]
The import script is nothing else than an updated version of the original one by [darknet](https://forums.epicgames.com/unreal-tournament-3/user-maps-mods-aa/beta-releases-aa/211996-blender-2-6x-add-on-export-psk-psa?t=747452). 
Which also used to have an export addon, not part of this repository though. It had been kept [kind-of-updated](https://github.com/blender/blender-addons/blob/v2.79b/io_export_unreal_psk_psa.py) (or at least fixed enough not to crash the whole program /s) until last year btw. 
But I fear porting it to 2.8 would still be difficult. 
Not sure if @lightnet may not be in the mood to help.

**Comment by mirh** [comment]
Ok, I tested it in 2.79.. and bones are kind of all over the place. 
Also, for some reason it only seems able to export inside the blender root folder, and only once you have selected the "filename from blender" option. 
Aside of that.. It could be worse I guess?
The mesh itself seems rock solid at least.

**Comment by GitHubCrazy1** [comment]
Thanks  - I tried as well over the past couple of weeks. I installed both the older 2.79 and 2.83 blender, exporting to the older blender if it worked. 

The bones appear to be all over the place for me as well.

**Comment by mirh** [comment]
I wonder if even [older](https://wiki.blender.jp/Extensions:2.6/Py/Scripts/Import-Export/Unreal_psk_psa) versions couldn't change anything

**Comment by mirh** [comment]
https://github.com/DarklightGames/io_export_psk_psa

**Comment by GitHubCrazy1** [comment]
Thanks a lot for your help  - I will take a look at the exporter.

Sorry about the slow reply. Had quite the upheaval in my life due to COVID 19. Am relocating right now or in the process of doing so.

**Comment by Blenux** [comment]
> Thanks a lot for your help - I will take a look at the exporter.
> 
> Sorry about the slow reply. Had quite the some changes in my life due to COVID 19. Am relocating right now or in the process of doing so. Long story short - had to make masks for hospitals, which is not my usual job, and am swapping jobs now because of fear of job loss.

Think we all can understand this, so don't stress.

**Comment by GitHubCrazy1** [comment]
Thanks - I finally just relocated.

Oddly despite downloading the addon zip and  installing it (and this may be because I'm new at Blender still), the exporter that Darklight Games has does not appear on Blender 2.83 in the addons list. 

I suspect that development may have halted. I saw that the author never responded to your bugs either.

Here is the linked in profile.
https://www.linkedin.com/in/colinbasnett

It looks like a one person operation - he may have simply moved on.

**Comment by mirh** [comment]
He's still committing to other related repos. 
Anyway, both the importer and exporter originate from the same codebase.. So it shouldn't be to hard to merge them back.

**Comment by GitHubCrazy1** [comment]
Thanks - I left him a message.  What file did you use to install? I tired adding both  the zip into Blender 2.83 and the Init file. Neither shows up in my Addons for some reason.

I tried downloading the whole io_export_psk_psa-master.zip and installing it with the addons, but no luck it seems. It does not show up in the list of addons. Something is very weird or I am doing something wrong.

**Comment by Blenux** [comment]
> 
> 
> Thanks - I left him a message. What file did you use to install? I tired adding both the zip into Blender 2.83 and the Init file. Neither shows up in my Addons for some reason.
> 
> I tried downloading the whole io_export_psk_psa-master.zip and installing it with the addons, but no luck it seems. It does not show up in the list of addons. Something is very weird or I am doing something wrong.


![Annotation 2020-07-05 152109](https://user-images.githubusercontent.com/25260650/86526136-354f7c80-bed3-11ea-9e7c-70e19b981828.png)

**Comment by mirh** [comment]
As usual, you download `io_export_psk_psa-master.zip`
Then unpack it, rezip just the `io_export_psk_psa` folder, and that's gonna be accepted.

**Comment by Blenux** [comment]
There is no need to rezip the io_export_psk_psa folder, put it directly into you scripts folder like above, refresh the addons tab and type uv and it will be there.

**Comment by GitHubCrazy1** [comment]
Thanks a lot for your help. I was able to get this issue resolved by reinstalling Blender. 

Once again, thanks a lot  - I really appreciate it.

**Comment by SolarisBravo** [comment]
Out of curiosity, what do you need .psk and .psa files for? UE4 doesn't support them and neither does whatever game they were "extracted" from (the format isn't actually used by any game, it's constructed from scratch by umodel).

**Comment by mirh** [comment]
Psk wasn't invented by umodel. At least until older UE3 versions, I think it was the [first](https://docs.unrealengine.com/udk/Two/BinaryFormatSpecifications.html) [party](https://docs.unrealengine.com/udk/Three/BinaryFormatSpecifications.html) mesh format. 
It's still supported in the UDK for example.
